src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -413,9 +413,28 @@ private DotnetHost DotnetHostInfo
                     CommandLineBuilderExtension commandLineBuilder = new CommandLineBuilderExtension();
                     AddCommandLineCommands(commandLineBuilder);
 
-                    if (string.IsNullOrEmpty(ToolPath) && string.IsNullOrEmpty(ToolExe))
+                    // ToolExe delegates back to ToolName if the override is not
+                    // set.
+                    // So, we can't check if it's unset, as that will recurse
```

Comment:
As my advisor used to say, "The act of recursion is to 'recur' -- 'recurse' is to curse, again." 😄